crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1857,9 +1874,9 @@ impl AuthorityPerEpochStore {
             .contains_key(key)?)
     }
 
-    pub fn check_consensus_messages_processed<'a>(
```

Comment:
I have updated this a bit so that there is no copy involved.  Now `is_any_tx_certs_consensus_message_processed` would have lifetime and pass-by-reference, then it owns `keys` that is later passed to and consumed by this function.